microsoft-edge/devtools-guide-chromium/console/console-dom-interaction.md

```diff
@@ -5,290 +5,345 @@ author: MSEdgeTeam
+
+1. Open a text editor, and then paste the copied image URLs into the editor.
+
+   ![The list of image URLs in a text editor](./console-dom-interaction-images/pasted-strings.png)
 
-<!-- mystery wiggly line preventer -->
 
+#### Clear the Console
 
-### Example: Getting all page headings and anchor URLs
+When running code in the **Console**, things can get messy.  You can clear the **Console** by using the `clear()` utility function.
 
-Ready for an even more complex example?  HTML webpages that are generated from Markdown tagging, like the present article, have automatic ID values for each heading, to allow you to deep-link directly to that section of the webpage.  For example, a `# New features` h1 heading in the Markdown source file becomes `<h1 id="new-features">New features</h1>` in the HTML file.
 
-To list all of the automatic headings to copy and paste:
+#### List the event listeners assigned to an element
 
-1. Open the **Console**.
+To list all event listeners that are assigned to an element, use the `getEventListeners()` utility function:
 
-1. Copy and paste the following code:
+1. Go to the webpage that you want to inspect. For example, open the [DevTools Console tool DOM interactions demo page](https://microsoftedge.github.io/Demos/devtools-console-dom-interactions/) in a new tab or window.
+
+1. Press **Ctrl+Shift+J** (Windows, Linux) or **Command+Option+J** (macOS).  The **Console** tool opens in DevTools, next to the webpage.
+
+1. Click the **Inspect** tool, and then in the rendered webpage, click one of the images to select it. The **Elements** tool is displayed, and the image element you selected is highlighted in the DOM tree:
+
+   ![The Elements tool, with the image element selected](./console-dom-interaction-images/inspect-image.png)
+
+1. Open the **Console** tool, enter the following code, and then press **Enter**:
 
    ```javascript
-   let out = '';
-   $$('#main [id]').filter(
-      elm => {return elm.nodeName.startsWith('H')}
-   ).forEach(elm => {
-      out += elm.innerText + "\n" +
-            document.location.href + '#' +
-            elm.id + "\n";
-   });
-   console.log(out);
+   getEventListeners($0);
    ```
-    
-   The result is text that contains content for each heading followed by the full URL that points to it.
 
-   ![Getting all the headings and the generated URLs from the webpage](./console-dom-interaction-images/console-dom-get-generated-headings.png)
+   In the above code snippet, `$0` is a shortcut to the currently selected `<img>` element in the **Elements** tool. The `getEventListeners()` utility function returns an object that contains all the event listeners that are assigned to the image:
 
+1. Expand the object, and then expand the only event listener in the list to see the listener's properties. The `listener` property shows the name of the function that's called when the event is triggered:
```

Comment:
```suggestion
1. Expand the object, and then expand the only event listener in the list to see the listener's properties. The `listener` property shows the name of the function that's called when the event is triggered (in this example, `showImage`):
```
ideally, per corp style guide aspiration, article should be clear w/o depending on pngs